ci.Jenkinsfile

```diff
@@ -18,17 +18,29 @@ pipeline {
   }
 
   parameters {
-      string(name: 'OPERATOR_YAML_RC_SHA', defaultValue: '')
+    string(name: 'OPERATOR_YAML_RC_SHA', defaultValue: '')
+    booleanParam(
+      name: 'FORCE_RUN_CI',
+      defaultValue: false,
+      description: 'To manually trigger a build, activate the FORCE_RUN_CI checkbox.'
+    )
   }
 
   stages {
     stage("Set RUN_CI") {
+      environment {
+        FILES_TO_CHECK = 'operator scripts collector ci.Jenkinsfile Makefile'
+      }
       steps {
         script {
-          env.RUN_CI = sh(
-            script: 'if [[ "$(git rev-parse origin/main)" == "${GIT_COMMIT}" ]]; then git diff --quiet --name-only --diff-filter=ADMR ${GIT_COMMIT}~..${GIT_COMMIT} -- operator scripts collector ci.Jenkinsfile Makefile && echo false || echo true; else git diff --quiet --name-only --diff-filter=ADMR origin/main..${GIT_COMMIT} -- operator scripts collector ci.Jenkinsfile Makefile && echo false || echo true; fi',
-            returnStdout: true
-          ).trim()
+          if (params.FORCE_RUN_CI) {
+            env.RUN_CI = 'true'
+          } else {
+            env.RUN_CI = sh(
+              script: './ci/jenkins/run-ci.sh -b ${GIT_COMMIT}~ -d ${GIT_COMMIT} -f "${FILES_TO_CHECK}"',
```

Comment:
This doesn't seem to maintain the current functionality of comparing ${GIT_COMMIT} with the main branch if on a PR branch. `origin/main..${GIT_COMMIT}` from the current functionality.

Reply:
The goal was to resolve the issue of PRs always triggering CI when you're making final changes to only update the dev-internal docs, since the current functionality compares the git commit from `origin/main` and all of the changes up to  `${GIT_COMMIT}`. Also, with the addition of being able to manually trigger CI, you could run CI if that is the desired goal.

Reply:
Changes have been made to maintain the current functionality of comparing PR branch with origin/main.